src/test/java/test/DynamicGraphTest.java

```diff
@@ -133,6 +140,55 @@ public void test2() {
   }
 
   @Test
+  public void test3() {
+          DynamicGraph<Node> dg = new DynamicGraph<>();
+          Node a = new Node("a");
+          Node b = new Node("b");
+          Node c = new Node("c");
+          dg.addNode(a);
+          dg.addNode(b);
+          dg.addNode(c);
+          dg.addEdge(1, a, b);
+          dg.addEdge(0, c, b);
+          dg.addEdge(0, b, a);
+
+          assertFreeNodesEquals(dg, b);
+
+          dg.setStatus(b, Status.RUNNING);
+          assertFreeNodesEquals(dg);
+
+          dg.setStatus(b, Status.FINISHED);
+          assertFreeNodesEquals(dg, a, c);
```

Comment:
It is not related to this pull request but I think `a` and `c` should not be free at the same time:
if `c -> b -> a` then an explicit edge should exist from `c` to `a`.
When `b` is finished then `a` should be free first and `c` only once `a` is finished.

Could you add a TODO comment about it? It will allow us to not be scared when the described issue will be closed (and the test modified)